microclient.go

```diff
@@ -0,0 +1,15 @@
+// microclient provides opinionated helpers for REST clients.
+package microclient
+
+import (
+	"context"
+
+	"github.com/go-resty/resty"
+)
+
+// Do takes a resty request function and executes it, adding metrics.
+func Do(ctx context.Context, requestFunc func(string) (resty.Response, error)) (resty.Response, error) {
```

Comment:
Just added the stub for now, this will be filled / changed / removed / obliterated in the next PR.